contrib/mesos/pkg/proc/types.go

```diff
@@ -71,4 +75,7 @@ type ErrorOnce interface {
 
 	// Send is non-blocking; it spins up a goroutine that reports an error (if any) that occurs on the error chan.
 	Send(<-chan error) ErrorOnce
+
+	// WaitFor returns true if an error is received within the specified duration, otherwise false
+	WaitFor(time.Duration) (error, bool)
```

Comment:
this is true. however, errors are the main subject of this interface: the
error that's returned here it not because WaitFor failed ... it's because
the error you were hoping to see within the duration actually showed up.
also it seemed (to me) more natural for callers to write: err, ok :=
WaitFor(someDuration)

On Sun, Jun 28, 2015 at 2:42 AM, Tomás Senart <notifications@github.com>
wrote:

> In contrib/mesos/pkg/proc/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/10457#discussion_r33419391>
> :
>
> > @@ -71,4 +75,7 @@ type ErrorOnce interface {
> >
> >  	// Send is non-blocking; it spins up a goroutine that reports an error (if any) that occurs on the error chan.
> >  	Send(<-chan error) ErrorOnce
> > +
> > +	// WaitFor returns true if an error is received within the specified duration, otherwise false
> > +	WaitFor(time.Duration) (error, bool)
>
> Errors are usually returned as the second argument, not the first.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/10457/files#r33419391>
> .
>